osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/monitoring/AbstractGetPowerQualityProfileHandler.java

```diff
@@ -591,11 +593,14 @@ private boolean obisMatches(final String captureObis, final String configObis) {
     return obisCodesToMatch.contains(captureObis);
   }
 
-  private String getScalerUnit(final CosemObject object) {
+  private String getScalerUnit(final DlmsConnectionManager conn, final CosemObject object)
+      throws ProtocolAdapterException {
     if (object.getClassId() == InterfaceClass.REGISTER.id()) {
-      return object.getAttribute(RegisterAttribute.SCALER_UNIT.attributeId()).getValue();
+      return this.dlmsHelper.getScalerUnitValue(
+          conn, object, RegisterAttribute.SCALER_UNIT.attributeId());
     } else if (object.getClassId() == InterfaceClass.EXTENDED_REGISTER.id()) {
-      return object.getAttribute(ExtendedRegisterAttribute.SCALER_UNIT.attributeId()).getValue();
+      return this.dlmsHelper.getScalerUnitValue(
```

Comment:
Should we supply a ExtendedRegisterAttribute.SCALER_UNIT to the method getScalerUnitValue, or can it be integrated in the method itself, so we don't need the parameter ExtendedRegisterAttribute.SCALER_UNIT.attributeId.

Reply:
Heb het gerefactored